cli/flox-rust-sdk/src/providers/services.rs

```diff
@@ -464,6 +468,89 @@ impl Iterator for ProcessComposeLogStream {
     }
 }
 
+/// Thin wrapper around the output of a single tailing [ProcessComposeLogReader].
+///
+/// This is an alternative to [ProcessComposeLogStream],
+/// which runs a single reader for a single process,
+/// and only collects the last up to `tail` lines.
+/// Unlike [ProcessComposeLogStream], this struct does not provide an [Iterator]
+/// to continuously read log lines.
+/// Because log lines are currently not timestamped,
+/// we can only meaningfully provide tail logs for a single process.
+pub struct ProcessComposeLogTail {
```

Comment:
nit: is this struct necessary? Seems like we could just have a function that returns `Vec<ProcessComposeLogLine>`